demisto_sdk/commands/format/update_generic.py

```diff
@@ -84,11 +84,21 @@ def set_version_to_default(self, location=None):
 
     def remove_unnecessary_keys(self):
         """Removes keys that are in file but not in schema of file type"""
-        arguments_to_remove = self.arguments_to_remove()
+        with open(self.schema_path, 'r') as file_obj:
+            schema = yaml.safe_load(file_obj)
         if self.verbose:
             print('Removing Unnecessary fields from file')
-        for key in arguments_to_remove:
-            self.data.pop(key, None)
+        self.recursive_remove_unnecessary_keys(schema.get('mapping'), self.data)
+
+    def recursive_remove_unnecessary_keys(self, schema, data):
+        """Recursively removes all the unnecessary fields in the file"""
+        data_keys = set(data.keys())
+        for key in data_keys:
+            if key not in schema.keys():
```

Comment:
if we are in verbose mode, maybe it is worth noting what keys we are removing in the format.
@roysagi WDYT?

Reply:
Could be a good feature for debugging on our side.

Reply:
Added (line 105)